js/mexc3.js

```diff
@@ -3638,186 +3645,169 @@ module.exports = class mexc3 extends Exchange {
+            throw new ArgumentsRequired (this.id + ' fetchDeposits() requires a network parameter when the currency is specified');
         }
+        // currently mexc does not have network names unified so for certain things we might need TRX or TRC-20
+        // due to that I'm applying the network parameter directly so the user can control it on its side
+        currency = this.currency (code);
+        request['currency'] = currency['id'] + '-' + rawNetwork;
         if (since !== undefined) {
             request['start_time'] = since;
         }
         if (limit !== undefined) {
+            if (limit > 1000) {
+                throw new ExchangeError ('This exchange supports a maximum limit of 1000');
+            }
             request['limit'] = limit;
         }
-        const response = await this.spot2PrivateGetAssetDepositList (this.extend (request, params));
+        const response = await this.spotPrivateGetCapitalDepositHisrec (this.extend (request, params));
         //
+        // [
         //     {
-        //         "code":200,
-        //         "data":{
-        //             "page_size":20,
-        //             "total_page":1,
-        //             "total_size":1,
-        //             "page_num":1,
-        //             "result_list":[
-        //                 {
-        //                     "currency":"USDC",
-        //                     "amount":150.0,
-        //                     "fee":0.0,
-        //                     "confirmations":19,
-        //                     "address":"0x55cbd73db24eafcca97369e3f2db74b2490586e6",
-        //                     "state":"SUCCESS",
-        //                     "tx_id":"0xc65a9b09e1b71def81bf8bb3ec724c0c1b2b4c82200c8c142e4ea4c1469fd789:0",
-        //                     "require_confirmations":12,
-        //                     "create_time":"2021-10-11T18:58:25.000+00:00",
-        //                     "update_time":"2021-10-11T19:01:06.000+00:00"
-        //                 }
-        //             ]
-        //         }
+        //         amount: '10',
+        //         coin: 'USDC-TRX',
+        //         network: 'TRX',
+        //         status: '5',
+        //         address: 'TSMcEDDvkqY9dz8RkFnrS86U59GwEZjfvh',
+        //         addressTag: null,
+        //         txId: '51a8f49e6f03f2c056e71fe3291aa65e1032880be855b65cecd0595a1b8af95b',
+        //         insertTime: '1664805021000',
+        //         unlockConfirm: '200',
+        //         confirmTimes: '203'
         //     }
+        // ]
         //
-        const data = this.safeValue (response, 'data', {});
-        const resultList = this.safeValue (data, 'result_list', []);
-        return this.parseTransactions (resultList, currency, since, limit);
+        return this.parseTransactions (response, currency, since, limit);
     }
 
     async fetchWithdrawals (code = undefined, since = undefined, limit = undefined, params = {}) {
```

Comment:
@AlgoQ also if possible add the "@see" property to the docstring in the methods you touched

Ex:
@see https://developer-pro.bitmart.com/en/spot/#get-k-line